core/src/main/java/io/grpc/internal/ManagedChannelImpl.java

```diff
@@ -721,6 +725,27 @@ public void run() {
         }).drain();
   }
 
+  @Override
+  public void shutdownTransports() {
+    channelExecutor
+        .executeLater(
+            new Runnable() {
+              @Override
+              public void run() {
+                if (shutdown.get()) {
+                  return;
+                }
```

Comment:
Add
```java
if (nameResolverStarted) {
  nameResolver.refresh();
}
```

? Otherwise the LB will still route new RPCs to the same **EAG** of the shutdown InternalSubchannel, which might have been resolved by the DNS for cellular nextwork.

Reply:
Done